### PR TITLE
Expose access to the druid regex

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made? 🤔

This will enable H2 to decouple from druid-tools: https://github.com/sul-dlss/happy-heron/blob/dbf2991b058bb668e74c26b00d49d287ac920882/lib/tasks/deposit.rake#L45

## How was this change tested? 🤨
ci

### CODE REVIEW

File: lib/cocina/models.rb [hunk 1]
```diff
@@ -153,5 +153,12 @@ def self.type_for(dyn)
       raise ValidationError, 'Type field not found'
     end
     private_class_method :type_for
+
+    def self.druid_regex
+      @druid_regex ||= begin
+        str = Openapi3Parser.load_file('openapi.yml').components.schemas['Druid'].pattern
```
Comment (justinlittman) [inline lib/cocina/models.rb:159]: Are you sure the overhead of parsing openapi is worth it vs. just duplicating it here?

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comment.

File: lib/cocina/models.rb (hunk 1 above)
Comment (jcoyne) [reply lib/cocina/models.rb:159]: Its memorized, so it should only be once per process. We can optimize it later if necessary.

Comment (ndushay) [reply lib/cocina/models.rb:159]: does openapi.yml strictly parse druids so the letters do not have vowels or the letter L ?

Comment (justinlittman) [reply lib/cocina/models.rb:159]: https://github.com/sul-dlss/cocina-models/blob/main/openapi.yml#L1045

**Pull request merged**